tests/dependency_test.cpp

```diff
@@ -0,0 +1,88 @@
+#include <gtest/gtest.h>
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include <algorithm>
+#include "../src/ws2st/ws2st.hpp"
+
+std::shared_ptr<ws2st::WashProgramMeta> ws2st_main(int argc, const char** argv) {
+    WashOptions programOptions = ws2st::args::parseCommandLine(argc, argv);
+    std::vector<std::string> allSources = ws2st::files::assembleSourceWithBackend(programOptions);
+    AllFiles = allSources;
+
+    ws2st::refactor::runRefactoring(programOptions);
+
+    return ws2st::program_meta;
+}
+
+std::vector<std::string> kernels = {
+    "VelocityUpdate",
+    "CalculateDensity",
+    "CalculatePressureForce",
+    "CalculateViscosity", 
+    "UpdatePositions",
+    "HandleCollisions",
+    "domain_sync"
+};
+
+std::unordered_map<std::string, std::vector<std::string>> reads_from_map {
+    { "VelocityUpdate", {"vel", "pos", "position"} },
+    { "CalculateDensity", {"pos"} },
+    { "CalculatePressureForce", {"density", "nearDensity", "pos", "vel"} },
+    { "CalculateViscosity", { "pos", "vel" } },
+    { "UpdatePositions", { "position", "vel" } },
+    { "HandleCollisions", { "position", "vel" } },
+    { "domain_sync", {} }
+};
+
+std::unordered_map<std::string, std::vector<std::string>> writes_to_map {
+    { "VelocityUpdate", {"vel", "pos"} },
+    { "CalculateDensity", {"density", "nearDensity"} },
+    { "CalculatePressureForce", {"pressure", "vel"} },
+    { "CalculateViscosity", { "viscosity", "vel" } },
+    { "UpdatePositions", { "position" } },
+    { "HandleCollisions", { "position", "pos", "vel" } },
+    { "domain_sync", {"density", "nearDensity", "vel"} }
+};
+
+template <typename T>
+bool checkVectorsContainTheSame(std::vector<T>& a, std::vector<T>& b, bool care_about_hangover = true) {
+    std::vector<T> bb = b;
+    std::vector<T> aa = a;
```

Comment:
If you change the function parameter from `std::vector<T>& a` to `std::vector<T>` then you don't need to explicit copy of `a` and `b`